### DISCUSSION TITLE
What are ServiceDiscoveryProvider settings in Kubernetes scenario?

### DISCUSSION THREAD

**Discussion by matsch01** [ThreeMammals/Ocelot#2006] [Q&A] [answered]
Can you please clarify the ServiceDiscoveryProvider Host, Port and Token are used for? 
- Is ok to rely on the default values provided in documentation?
- What are common scenarios when it's required/recommended to change them from what is shown in the documentation?
- What are the guidelines for changing them? How does one generate a new Token?

**Comment by gionapaolini** [comment]
I agree with you the documentation for this is not clear.
Also I don't understand why it won't work by using the service name as host, that is the way in which services communicates in kubernetes (at least in the same namespace).

**Comment by raman-m** [comment, marked as answer]
Dear students,

### Our SD docs
- [Kubernetes aka K8s](https://ocelot.readthedocs.io/en/latest/features/kubernetes.html)
- [Service Discovery](https://ocelot.readthedocs.io/en/latest/features/servicediscovery.html)

**Reply by raman-m** [reply]
Especially you would be interested in settings
- [Kube default provider](https://ocelot.readthedocs.io/en/latest/features/kubernetes.html#kube-default-provider)
